### DISCUSSION TITLE
How project based common classes needs to be configured with spring-modulith

### DISCUSSION THREAD

**Discussion by metealpaslan** [spring-projects/spring-modulith#82] [Q&A] [answered]
What is common practice keeping some project based common classes such interceptors, filters, message converters or some utility classes which might be used by all modules in same monolith project structure while using spring-modulith?

**Comment by danstooamerican** [comment]
We need something like this as well.

**Comment by gkislin** [comment]
The same question. Common module definitely contains sub-module and all of them must be shared.

**Comment by odrotbohm** [comment, marked as answer]
Most projects I know of simply declare a `core` module and register that as a shared module via the `@Modulithic` annotation. See an example of that [here](https://github.com/st-tu-dresden/salespoint/blob/5c81e31c820eefa6e75837beb3cbd7f6778dbc56/src/main/java/org/salespointframework/EnableSalespoint.java#L58-L62).

**Reply by nikimicallef** [reply]
Thank you for this. It would be nice if we can somehow exclude these modules from the documentation/PlantUML diagram.

**Reply by odrotbohm** [reply]
`DiagramOptions` has `….withComponentFilter(…)` that allows selecting which modules to include into the diagrams created.

**Reply by nikimicallef** [reply]
Lovely, thank you! I managed to make it work :)
